test/Microsoft.SqlTools.ServiceLayer.UnitTests/EditData/RowDeleteTests.cs

```diff
@@ -200,6 +200,36 @@ public async Task RevertCell()
             Assert.Throws<InvalidOperationException>(() => rd.RevertCell(0));
         }
 
+        [Fact]
+        public async Task GetVerifyQuery(){
+             // Setup: Create a row delete
+            Common.TestDbColumnsWithTableMetadata data = new Common.TestDbColumnsWithTableMetadata(false, false, 0, 0);
+            var rs = await Common.GetResultSet(data.DbColumns, false);
+            RowDelete rd = new RowDelete(0, rs, data.TableMetadata);
+            int expectedKeys = 3;
+
+            // If: I generate a verify command
+            String verifyCommand = rd.GetVerifyScript();
+
+             // Then:
+            // ... The command should not be null
+            Assert.NotNull(verifyCommand);
+
+            // ... It should be formatted into an where script
+            string regexTest = @"SELECT COUNT \(\*\) FROM (.+) WHERE (.+)";
```

Comment:
So this is better than nothing, but the point of the testing would be to verify that an error is thrown if the query comes back with more or less than 1 row. 

What kind of issues were you having? Can you show the code you had that you were having issues with?

Reply:
The issue I'm currently having is creating a dataset that has multiple rows (with the same data) and deleting it with the appropriate response, I can't seem to get the query to execute properly with the TestDbDataReader (as it simply returns the next row unaltered). I would like to see an example of a test that actually executes queries on a dataset that is altered.

Reply:
For reference here is my original code:
`
[Fact]
        public async Task GetVerifyQuery(){
            // Setup: create result set to be worked on.
            Common.TestDbColumnsWithTableMetadata data = new Common.TestDbColumnsWithTableMetadata(false, false, 0, 0);
            var rs = await Common.GetResultSet(data.DbColumns, false);
            var testResultSet = PreCreateRowResultSet(data.DbColumns, rs.GetRow(0));
            var newRowReader = new TestDbDataReader(new [] {testResultSet}, false);
            RowCreate rc = new RowCreate(1, rs, data.TableMetadata);
            await rc.ApplyChanges(newRowReader);


            var x = rs.GetRow(0);
            var y = rs.GetRow(1);
            Assert.Equal(2, rs.RowCount);

            RowDelete rd = new RowDelete(1, rs, data.TableMetadata);
            testResultSet = PostCreateRowResultSet(data.DbColumns, rs.GetRow(0));
            // ... Mock db connection for building the command
            var mockConn = new TestSqlConnection(new [] {testResultSet});
            // If: I attempt to get a command for the edit
            DbCommand cmd = rd.GetCommand(mockConn);

            Assert.True(true);

        }
`